src/engraving/dom/fret.h

```diff
@@ -125,6 +125,13 @@ class FretUndoData
     BarreMap m_barres;
     MarkerMap m_markers;
     DotMap m_dots;
+
+    int m_strings = 0;
+    int m_frets = 0;
```

Comment:
What about these members: m_userMag, m_numPos, m_showFingering, m_fingering? Perhaps we should store a copy of the m_diagram object and use the = operator to restore the state (or store all the members in a separate struct and reuse it in FretDiagram and FretUndoData). But if that's difficult to implement, the current solution is fine for now